cores/esp8266/core_esp8266_main.cpp

```diff
@@ -188,31 +193,25 @@ void init_done() {
 
 */
 
-#ifdef NO_EXTRA_4K_HEAP
-/* this is the default NONOS-SDK user's heap location */
-cont_t g_cont __attribute__ ((aligned (16)));
-#endif
-
-extern "C" void ICACHE_RAM_ATTR app_entry(void)
+extern "C" void ICACHE_RAM_ATTR app_entry_redefinable(void) __attribute__((weak));
+extern "C" void ICACHE_RAM_ATTR app_entry_redefinable(void)
 {
-#ifdef NO_EXTRA_4K_HEAP
-
-    /* this is the default NONOS-SDK user's heap location */
-    g_pcont = &g_cont;
-
-#else
-
     /* Allocate continuation context on this SYS stack,
        and save pointer to it. */
     cont_t s_cont __attribute__((aligned(16)));
     g_pcont = &s_cont;
 
-#endif
-
     /* Call the entry point of the SDK code. */
     call_user_start();
 }
 
+static void ICACHE_RAM_ATTR app_entry_custom (void) __attribute__((weakref("app_entry_redefinable")));
+
+extern "C" void ICACHE_RAM_ATTR app_entry (void)
+{
+    return app_entry_custom();
```

Comment:
Can't we call app_entry_redefinable() directly here? or is that what is meant about the extra level of indirection in the comments above?

Reply:
See comment above: tried that 3 times with different context with no luck. It is because (maybe) `app_entry` is defined as an ENTRY in ldscripts.
```
tools/sdk/ld/eagle.app.v6.common.ld.h:ENTRY(app_entry)
```